controllers/remote/suite_test.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package remote
+
+import (
+	"context"
+	"path/filepath"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog"
+	"k8s.io/klog/klogr"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest/printer"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha3"
+	// +kubebuilder:scaffold:imports
+)
+
+// These tests use Ginkgo (BDD-style Go testing framework). Refer to
+// http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
+
+func init() {
+	klog.InitFlags(nil)
+	logf.SetLogger(klogr.New())
+
+	// Register required object kinds with global scheme.
+	_ = clusterv1.AddToScheme(scheme.Scheme)
+}
+
+const (
+	timeout = time.Second * 10
+)
+
+var (
+	cfg     *rest.Config
+	testEnv *envtest.Environment
+	ctx     = context.Background()
+)
+
+func TestAPIs(t *testing.T) {
+	RegisterFailHandler(Fail)
+
```

Comment:
```suggestion
   
   flag.Set("logtostderr", "false")
   flag.Parse()
   klog.SetOutput(GinkgoWriter)
   
```

the magic incantation to get klog to stop logging to stdout/stderr

Reply:
Today I learned! Thank you, will update! Does this mean Ginkgo only outputs the log lines for the failed tests?

Reply:
Yeah exactly, without this if you run it with the `ginkgo` CLI you get a bunch of controller logs even on success. `go test` does something weird where it hides stdout/stderr somehow, but with both you now get the logs only if they fail.